tools/codegen/dest/register_dispatch_key.py

```diff
@@ -336,6 +345,22 @@ def gen_one(self, f: NativeFunction) -> Optional[str]:
         if self.target is Target.REGISTRATION and not self.selector.is_native_function_selected(f):
             return None
 
+        # TODO: Now, there is something interesting going on here.  In the code below,
+        # we generate DefaultBackend implementations of functional and inplace
+        # based on the out implementation.  But in fact, out is definable by
+        # functional too (just not very efficiently), and this is honestly the
+        # MORE likely situation for a backend implementor.  How do we pick?
+        # Well, taking a page from Haskell type classes and default methods,
+        # we could conceivably register a circular definition (out in terms
+        # of functional, and functional in terms of out) and just require
+        # someone to implement one or the other.  We'd have to do a little bit
+        # of work to not register one of these "weak" definitions unless there
+        # is a strong definition somewhere in the DAG!  So it's not implemented yet.
+        if self.dispatch_key == DispatchKey.DefaultBackend and f.func.kind() is SchemaKind.out:
+            # Never generate a default implementation for out, that's what you
+            # have to define as a backend implementor
+            return None
```

Comment:
Maybe I'm missing something- Doesn't the previous assert prevent you from hitting this case? You have an assert near the top of the file to check that an `out` variant native_function that's marked `structured` doesn't have a `DefaultBackend` key registered.

Reply:
They're asserting different things. 

```
            assert self.dispatch_key not in g.out.dispatch, \
                "Do not explicitly specify DefaultBackend dispatch key on structured " \
                "functions, they will be automatically generated for you"
```

asserts that in the dispatch table (in `native_functions.yaml`), you're not allowed to have a DefaultBackend entry. But even if you DON'T have a DefaultBackend entry, we will still call into codegen (something similar happens for Meta, where you never write Meta in the dispatch table, but Meta always gets generated. We don't iterate over the dispatch dictionary to figure out what to codegen!)